frontend/src/components/modules/AdvancementLevel/AdvancementLevel.tsx

```diff
@@ -0,0 +1,82 @@
+'use client';
+import { useState } from 'react';
+import Markdown from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+import { ChevronRightIcon } from '@app/static/icons/ChevronRightIcon';
+import { AccordionCard } from '@app/components/common/AccordionCard';
+import { AccordionList } from '@app/components/common/AccordionList';
+import { Modal } from '@app/components/common/Modal';
+import { AdvancementLevelProps } from './AdvancemetLevel.interface';
+
+export const AdvancementLevel: React.FC<AdvancementLevelProps> = ({ data, showVerticalLine }) => {
+  const { advancementLevel, description, categories, projects } = data;
+  const [open, setOpen] = useState(false);
+  const [isModalOpen, setModalOpen] = useState(false);
+
+  const shouldBeExpandedByDefault = Object.keys(categories).length === 1;
+
+  const toggleOpen = () => setOpen((prevState) => !prevState);
+
+  const openModal = () => setModalOpen(true);
+
+  const hideModal = () => setModalOpen(false);
+
+  return (
+    <div className="flex flex-row">
+      <div className="flex flex-col items-center relative">
+        <div
+          className="mt-4 mb-2 bg-blue-800 rounded-full w-6 h-6 flex items-center justify-center hover:opacity-50"
+          onClick={toggleOpen}
+        >
+          <ChevronRightIcon className={`text-white w-3.5 h-3.5 ${!open ? 'rotate-90' : '-rotate-90'}`} />
+        </div>
+        {showVerticalLine && <div className="w-[1.5px] bg-blue-800 absolute top-12 left-3 h-[calc(100%-40px)]" />}
+      </div>
+      <div
+        className={`p-4 ml-2 flex flex-col gap-4 mb-4 w-full rounded-lg ${!open && 'hover:bg-navy-50'}`}
+        onClick={open ? undefined : toggleOpen}
```

Comment:
```suggestion
        onClick={openModal}
```
for simplicity, will it work here?

Reply:
No, because toggleOpen and openModal change different states. I renamed them to avoid confusion, so `open` was renamed to `accordionOpen`.